pandas/io/formats/style.py

```diff
@@ -340,6 +343,319 @@ def to_excel(
+        {} & {} & {Integers} & {Floats} & {Strings} \\\\
+        \\multirow[c]{2}{*}{L0} & ix1 & \\$1 & 2.200 & DOGS \\
+         & ix2 & \\$3 & 4.400 & CATS \\\\
+        L1 & ix3 & \\$2 & 6.600 & COWS \\\\
+        \\end{tabular}
+        """
+        table_selectors = (
+            [style["selector"] for style in self.table_styles]
+            if self.table_styles is not None
+            else []
+        )
+
+        if column_format is not None:
+            # add more recent setting to table_styles
+            self.set_table_styles(
+                [{"selector": "column_format", "props": f":{column_format}"}],
+                overwrite=False,
+            )
+        elif "column_format" in table_selectors:
+            pass  # adopt what has been previously set in table_styles
+        else:
+            # create a default: set float, complex, int cols to 'r' ('S'), index to 'l'
+            numeric_cols = list(self.data.select_dtypes(include=[np.number]).columns)
+            numeric_cols = list(self.columns.get_indexer_for(numeric_cols))
+            column_format = "" if self.hidden_index else "l" * self.data.index.nlevels
+            for ci, _ in enumerate(self.data.columns):
+                if ci not in self.hidden_columns:
+                    column_format += (
+                        ("r" if not siunitx else "S") if ci in numeric_cols else "l"
+                    )
+            self.set_table_styles(
+                [{"selector": "column_format", "props": f":{column_format}"}],
+                overwrite=False,
+            )
+
+        if position:
+            self.set_table_styles(
+                [{"selector": "position", "props": f":{position}"}],
+                overwrite=False,
+            )
+
+        if position_float:
+            self.set_table_styles(
+                [{"selector": "position_float", "props": f":{position_float}"}],
+                overwrite=False,
+            )
+
+        if hrules:
+            self.set_table_styles(
+                [
+                    {"selector": "toprule", "props": ":toprule"},
+                    {"selector": "midrule", "props": ":midrule"},
+                    {"selector": "bottomrule", "props": ":bottomrule"},
+                ],
+                overwrite=False,
+            )
+
+        if label:
+            self.set_table_styles(
+                [{"selector": "label", "props": f":{{{label.replace(':', '§')}}}"}],
+                overwrite=False,
+            )
+
+        if caption:
+            self.set_caption(caption)
+
+        if sparsify is not None:
+            with pd.option_context("display.multi_sparse", sparsify):
+                latex = self._render_latex(
+                    multirow_align=multirow_align, multicol_align=multicol_align
+                )
+        else:
+            latex = self._render_latex(
+                multirow_align=multirow_align, multicol_align=multicol_align
+            )
```

Comment:
could maybe use `contextlib.nullcontext` or pass through `sparsify` to `._render_latex` and handle there?

Reply:
yes good idea, would prefer to do this more generically as a follow-on: #41142 (logged an issue)